src/led.rs

```diff
@@ -0,0 +1,34 @@
+/// Power LED of the flashlight. [pwn] represents the duty cycle.
+pub trait Led {
+    fn set_pwm(&mut self, pwm: u8);
+    fn get_pwm(&self) -> u8;
+}
+
+/// Led which resides in memory, for simulation or testing
+pub struct DummyLed {
+    pwm: u8
+}
+
+impl DummyLed {
+    /// Factory function to create a dummy LED
+    pub fn create(pwm: u8) -> Box<dyn Led> {
+        return Box::new(DummyLed { pwm });
+    }
+}
+
+/// Higher level factory function to create a dummy LED

```

Comment:
It looks like factory functions aren't so idiomatic for Rust, but this may be what we need:
https://github.com/rust-unofficial/patterns/blob/master/patterns/builder.md